### PR TITLE
GEODE-4414: Support explicit nulls in geode's protobuf protocol

### PR DESCRIPTION
**Pull request by WireBaron**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/ProtobufSerializationService.java [hunk 1]
```diff
@@ -120,7 +122,7 @@ public Object decode(BasicTypes.EncodedValue encodedValue) throws EncodingExcept
         return encodedValue.getStringResult();
       case JSONOBJECTRESULT:
         return jsonPdxConverter.decode(encodedValue.getJsonObjectResult());
-      case VALUE_NOT_SET:
+      case NULLRESULT:
```
Comment (PivotalSarge) [inline geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/ProtobufSerializationService.java]: Should we still handle `VALUE_NOT_SET`? Or has that gone the way of the dodo?

Comment (WireBaron) [reply geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/ProtobufSerializationService.java]: Well, it's part of the protobuf spec, so we can't really get rid of it, but we're no longer going to support it.  An EncodedValue without a type set will be considered incorrectly encoded and throw an exception.

File: geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/ProtobufSerializationService.java [hunk 2]
```diff
@@ -36,58 +37,59 @@ public ProtobufSerializationService() {}
...
+            break;
+          }
+          case STRING: {
+            builder.setStringResult((String) value);
+            break;
+          }
+          case PDX_OBJECT: {
+            builder.setJsonObjectResult(jsonPdxConverter.encode((PdxInstance) value));
+            break;
+          }
```
Comment (upthewaterspout) [inline geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/ProtobufSerializationService.java]: This isn't due to your change, but should we have a default: in this switch statement to throw an exception if we missed a type?

Comment (WireBaron) [reply geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/ProtobufSerializationService.java]: Oh wow, apparently java have an implicit noop default.  I guess I was expecting a implicit throwing default, which doesn't necessarily make more sense, but yeah, I'll add this as explicit case.

File: geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/ProtobufSerializationService.java [hunk 3]
```diff
@@ -36,58 +37,59 @@ public ProtobufSerializationService() {}
...
-        }
-        case STRING: {
-          builder.setStringResult((String) value);
-          break;
-        }
-        case PDX_OBJECT: {
-          builder.setJsonObjectResult(jsonPdxConverter.encode((PdxInstance) value));
-          break;
+    if (value == null) {
+      builder.setNullResult(NullValue.NULL_VALUE);
```
[the remark this answers is not in the document]
Comment (WireBaron) [reply geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/ProtobufSerializationService.java]: Sure. I had thought this method would be cleaner with a single return call, but the resulting giant else block is even worse.

File: geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/operations/GetRequestOperationHandler.java [hunk 4]
```diff
@@ -66,6 +66,9 @@
 
       BasicTypes.EncodedValue encodedValue = serializationService.encode(resultValue);
       return Success.of(RegionAPI.GetResponse.newBuilder().setResult(encodedValue).build());
+    } catch (NullPointerException ex) {
```
[the remark this answers is not in the document]
Comment (WireBaron) [reply geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/operations/GetRequestOperationHandler.java]: Adding null checks to all these decode calls.

**Pull request merged**